spec/element_base_spec.rb

```diff
@@ -0,0 +1,94 @@
+require 'spec_helper'
+
+RSpec.describe Kitchen::ElementBase do
+
+  let(:element) do
+    book_containing(html:
+      one_chapter_with_one_page_containing(
+        <<~HTML
+          <div data-type="example" class="class1" id="div1">
+            <p>This is a paragraph.</p>
+          </div>
+        HTML
+      )
+    )
+  end
+
+  describe '#has_class?' do
+    it 'returns true if element has given class' do
+      div_element = element.search('div[data-type="example"]').first
```

Comment:
you can use the `first!` method instead: `element.first!('div[data-type="example"]')`  Ditto elsewhere.

`first!` blows up if the element isn't found, which is better here than returning `nil`.  The element should definitely be there, so better to know faster if you mistyped the search terms.